### PR TITLE
docs: add hugepagesavailable node condition details

### PR DESCRIPTION
**Pull request by jangseon-ryu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9983 https://github.com/longhorn/longhorn-manager/pull/3350

#### What this PR does / why we need it:
Added documentation for the HugePagesAvailable node condition
Explains the purpose of the condition in ensuring HugePages (2Mi) configuration for Longhorn v2 data engine

### CODE REVIEW

**Review by derekbit — APPROVED** [review]
Technically, LGTM.

Waiting for doc engineers' approvals cc

**Review by COLDTURNIP — APPROVED** [review]

**Review by innobead — APPROVED** [review]
LGTM. Thanks for the contributions including code and doc. 

Waiting for doc engineer review before merging.

**Review by jillian-maroket — APPROVED** [review]

**Review by derekbit — APPROVED** [review]

**Pull request merged**
